starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,76 @@
+const express = require("express");
+const authRoutes = express.Router();
+const User = require("../models/user");
+const bcrypt = require("bcryptjs");
+const bcryptSalt = 10;
+
+/* GET home page */
+authRoutes.get("/", (req, res, next) => {
+  res.render("index");
+});
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+authRoutes.get("/login", (req, res, next) => {
+  res.render("auth/login");
+});
+authRoutes.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+  if (username === "" || password === "") {
+    res.render("auth/signup", {
+      errorMessage: "Indicate a username and a password to sign up"
+    });
+    return;
+  }
+  User.findOne({ username: username }, "username", (err, user) => {
```

Comment:
This `"username", ` shouldn't be here... 🐛 
Should be something like this 👍 --> `User.findOne({ username: username }).then(data => {`

Reply:
Well I Search in Moongose doc, and you can also use the "username" as a parameter in the search. You can use both 😋